src/Features/Core/Portable/EditAndContinue/IEditAndContinueWorkspaceService.cs

```diff
@@ -30,5 +32,46 @@ internal interface IEditAndContinueWorkspaceService : IWorkspaceService, IActive
 
         ValueTask<bool?> IsActiveStatementInExceptionRegionAsync(Solution solution, ManagedInstructionId instructionId, CancellationToken cancellationToken);
         ValueTask<LinePositionSpan?> GetCurrentActiveStatementPositionAsync(Solution solution, SolutionActiveStatementSpanProvider activeStatementSpanProvider, ManagedInstructionId instructionId, CancellationToken cancellationToken);
+
+        // Used by non-VS clients
+        ValueTask<(ManagedModuleUpdates2 Updates, ImmutableArray<DiagnosticData> Diagnostics)> EmitSolutionUpdate2Async(Solution solution, CancellationToken cancellationToken);
+        void StartEditSession(out ImmutableArray<DocumentId> documentsToReanalyze);
+    }
+
+    internal readonly struct ManagedModuleUpdates2
```

Comment:
❔ Why does `ManagedModuleUpdates` not work? It's defined in a netstandard2.0 assembly and appears to be portable.

Reply:
It's defined as an internal type inside the VS.Contracts assembly. It was working fine until last week when 16.9 released at which point the type because no longer accessible to dotnet-watch since it does not have IVT access to that assembly.

Reply:
Ah, cool. So the ideal resolution here is to keep the Roslyn portion the same, but add a new external access assembly (I can help with this tomorrow). Inside the external access assembly, we create wrappers to these types which are accessible to another tool.

Reply:
Thanks @sharwell. I was able to set up an ExternalAccess proejct and give watch access to it thru the wrappers there. I'm not sure what the right way is to access the new API via MEF is. Is making the new API a IWorkspaceService the right approach? If so, any idea what I need to do to access it. Here's what I tried so far: https://github.com/dotnet/sdk/pull/16313/files#diff-8659ed832016f6727079193223db40704a36b9f81bf17ceba40b57df2c4e857bR31-R32